### PR TITLE
feat(absolute-time-picker-dropdown)!: add new dropdown component and single date functionality

### PR DESCRIPTION
**Pull request by GoncaloCamaz**
**Breaking change**

- Renamed prop selectedRangeDates to selectedDates and event onSelectedRangeDatesChange to onSelectedDatesChange
- Renamed type SelectedRangeTimestamp to SelectedTimestamp
- Removed default max date

New Functionality:
- Add single date mode on absolute-time-picker component

Screenshot 2024-02-16 at 11 38 43

Screenshot 2024-02-16 at 11 39 35

### CODE REVIEW

File: packages/ui-components/src/components/absolute-time-picker-dropdown/absolute-time-picker-dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+			const date = createTimestampInTimezoneFromFormattedDate(from, timezoneName, CALENDAR_DATE_TIME_MASK);
+			this.selectedDateState = [date];
+		} else {
+			const dateFrom = createTimestampInTimezoneFromFormattedDate(from, timezoneName, CALENDAR_DATE_TIME_MASK);
+			const dateTo = !isEmpty(to) ? createTimestampInTimezoneFromFormattedDate(to, timezoneName, CALENDAR_DATE_TIME_MASK) : undefined;
+			this.selectedDateState = [dateFrom, dateTo];
+		}
+	};
+
+	private getAbsoluteRange = (): string[] => {
```
Comment (djcouto) [inline packages/ui-components/src/components/absolute-time-picker-dropdown/absolute-time-picker-dropdown.tsx:103]: This too.

File: packages/ui-components/src/components/time-picker/time-picker.tsx [hunk 2]
```diff
@@ -290,9 +288,9 @@ export class KvTimePicker implements ITimePicker, ITimePickerEvents {
 	// Components config methods
 	private isApplyButtonDisabled() {
```
Comment (djcouto) [inline packages/ui-components/src/components/time-picker/time-picker.tsx:289]: This method could use some refactor, don't you agree?
Maybe something of this sort:

```ts
        private isApplyButtonDisabled() {
		if (!this.selectedTimeState) {
			return true;
		}

		if (!this.selectedTimeState) {
			return false;
		}

		if (this.selectedTimeState.range) {
			const { range: currentRange } = this.selectedTimeState;
			const { range: newRange } = this.selectedTimeOption;

			if (currentRange.length === FULL_RANGE_SIZE && hasRangeChanged(currentRange, newRange) && validateNewRange(newRange)) {
				return false;
			}
		}

		if (this.selectedTimeState.timezone?.name !== this.selectedTimeOption.timezone?.name) {
			return false;
		}

		return true;
	}
```

**Review by djcouto — CHANGES_REQUESTED** [review]
Nice work 👏

**Review by djcouto — APPROVED** [review]
🚀

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.types.ts [hunk 3]
```diff
@@ -1,13 +1,20 @@
 import { EventEmitter } from '@stencil/core';
 import { SelectedRange } from '../../types';
 
+export enum EAbsoluteTimePickerMode {
+	Point = 'point',
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.types.ts:5]: `Single` mode seems more clear

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.config.ts [hunk 4]
```diff
@@ -6,7 +8,7 @@ export const DATE_INPUT_MASK = 'DD-MM-YYYY';
 export const CALENDAR_MASK = 'YYYY-MM-DD';
 export const CALENDAR_DATE_TIME_MASK = 'YYYY-MM-DD HH:mm:ss';
 export const CALENDAR_INPUT_MIN_DATE = '01-01-2018 00:00:00';
-export const CALENDAR_INPUT_MAX_DATE = '31-12-2050 23:59:59';
-export const MAX_INPUT_YEAR = 2050;
+export const CALENDAR_INPUT_MAX_DATE = dayjs('31-12-2050 23:59:59', DATETIME_INPUT_MASK).add(50, 'year').format(DATETIME_INPUT_MASK);
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.config.ts:11]: Should these configs be removed in the future? If we don't provide a min or max on the calendar we should render until infinite.

File: packages/react-ui-components/src/stories/Components/AbsoluteTimePickerDropdown.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+	const onSelectedDatesChange = ({ detail: newDates }: CustomEvent<SelectedTimestamp>) => {
+		setSelectedDates(newDates);
+	};
+
+	return <KvAbsoluteTimePickerDropdown {...args} selectedDates={selectedDates} onSelectedDatesChange={onSelectedDatesChange} />;
+};
+
+const SingleDateTemplate: ComponentStory<typeof KvAbsoluteTimePickerDropdown> = args => {
+	const [selectedDates, setSelectedDates] = useState<SelectedTimestamp>([1681319856833]);
```
Comment (miguelpinto98) [inline packages/react-ui-components/src/stories/Components/AbsoluteTimePickerDropdown.stories.tsx:38]: This `1681319856833` should be a constant and also the end date

**Review by miguelpinto98 — APPROVED** [review]
🚀  Only some suggestion to discuss

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.types.ts (hunk 3 above)
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.types.ts:5]: `Single` perhaps?

Comment (GoncaloCamaz) [reply packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.types.ts:5]: Miguel has already pointed that out

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.config.ts (hunk 4 above)
Comment (GoncaloCamaz) [reply packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.config.ts:11]: Actually we dont need the date '31-12-2050 23:59:59', we could use today and add 50 years. But I guess i can ignore this limits and use a prop instead.

**Review by pedrorebelocosta — APPROVED** [review]

File: packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.config.ts (hunk 4 above)
Comment (GoncaloCamaz) [reply packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.config.ts:11]: I can remove this configs and use the props instead.

**Pull request merged**